openedx/core/djangoapps/content_libraries/serializers.py

```diff
@@ -57,12 +57,22 @@ class ContentLibraryPermissionLevelSerializer(serializers.Serializer):
     access_level = serializers.ChoiceField(choices=ContentLibraryPermission.ACCESS_LEVEL_CHOICES)
 
 
+class ContentLibraryAddPermissionByEmailSerializer(serializers.Serializer):
+    """
+    Serializer for adding a new user and granting their access level via their email address.
+    """
+    access_level = serializers.ChoiceField(choices=ContentLibraryPermission.ACCESS_LEVEL_CHOICES)
+    email = serializers.EmailField()
+
+
 class ContentLibraryPermissionSerializer(ContentLibraryPermissionLevelSerializer):
     """
     Serializer for a ContentLibraryPermission object, which grants either a user
     or a group permission to view a content library.
     """
     user_id = serializers.IntegerField(source="user.id", allow_null=True)
+    email = serializers.EmailField(source="user.email", read_only=True)
+    username = serializers.EmailField(source="user.username", read_only=True)
```

Comment:
@bradenmacdonald Confirming if it's okay to keep these in here, in light of the above discussion

Reply:
@mavidser @Kelketek Based on @robrap's input above, I think that this API will need to remove `user_id` from the REST API (it can stay in the python API) and identify users only by usernames (and allow adding them via username or email). Because it's a product requirement to display usernames here, and so if we also include `user_id` we'd be leaking the relationship between the PII (username/email) and the user ID.

I checked and confirmed that LabXchange is not using the `team` endpoints, so we should be fine to change the REST API.

Reply:
Thanks. FYI: @nasthagiri